generator/src/test/java/com/linkedin/pegasus/generator/TestDataSchemaParser.java

```diff
@@ -92,6 +102,57 @@ public void testParseFromJarFile(String pegasusFilename, String[] expectedSchema
     parseResult.getSchemaAndLocations().values().forEach(loc -> assertEquals(loc.getSourceFile().getAbsolutePath(), jarFile));
   }
 
+  @DataProvider(name = "entityRelationshipInputFiles")
+  private Object[][] createResolverWithExtensionDirs()
+  {
+    return new Object[][]
+        {
+            {
+                new String[]{
+                    "extensions/Bar.pdl",
+                    "extensions/FooExtension.pdl",
+                    "pegasus/Foo.pdl"
+                },
+                new String[]{
+                    "FooExtension",
+                    "Bar"
+                }
+            },
+        };
+  }
+
+  @Test(dataProvider = "entityRelationshipInputFiles")
+  public void testSchemaFilesInExtensionPath(String[] files, String[] expectedExtensions) throws Exception
+  {
+    String tempDirectoryPath = _tempDir.getAbsolutePath();
+    String jarFile = tempDirectoryPath + FS + "test.jar";
+    String schemaDir = pegasusDir + FS + "entity-relationship";
```

Comment:
shall we make the file path as camelcase? Also update the test files path.

Reply:
done